rust/osm2lanes-web/src/draw.rs

```diff
@@ -1,131 +1,148 @@
-    let canvas_height = canvas_height as f64;
-    let grassy_verge = 10.0;
-    let asphalt_buffer = 10.0;
-    let lane_width = 80.0;
     rc.fill(
         Rect::new(
-            grassy_verge,
+            scale.scale(grassy_verge),
             0.0,
-            (grassy_verge + asphalt_buffer) + (lanes.len() as f64 * lane_width) + asphalt_buffer,
+            scale.scale(grassy_verge + asphalt_buffer + road.total_width() + asphalt_buffer),
             canvas_height,
         ),
         &Color::BLACK,
     );
-    let x = grassy_verge + asphalt_buffer - 5.0;
-    rc.stroke(
-        Line::new(
-            Point { x, y: 0.0 },
-            Point {
-                x,
-                y: canvas_height as f64,
-            },
-        ),
-        &Color::WHITE,
-        1.0,
-    );
-    let x = (grassy_verge + asphalt_buffer) + (lanes.len() as f64 * lane_width) + 5.0;
-    rc.stroke(
-        Line::new(
-            Point { x, y: 0.0 },
-            Point {
-                x,
-                y: canvas_height,
-            },
-        ),
-        &Color::WHITE,
-        1.0,
-    );
-    for (idx, lane) in lanes.iter().enumerate() {
-        // asphalt
-        rc.fill(
-            Rect::new(
-                (grassy_verge + asphalt_buffer) + (idx as f64 * lane_width),
-                0.0,
-                (grassy_verge + asphalt_buffer) + ((idx + 1) as f64 * lane_width),
-                canvas_height,
-            ),
-            &Color::BLACK,
-        );
-        // left line
-        let x = (grassy_verge + asphalt_buffer) + (idx as f64 * lane_width);
-        rc.stroke(
-            Line::new(
-                Point { x, y: 0.0 },
-                Point {
-                    x,
-                    y: canvas_height,
-                },
-            ),
-            &Color::WHITE,
-            1.0,
-        );
-        // right line
-        let x = (grassy_verge + asphalt_buffer + lane_width) + (idx as f64 * lane_width);
-        rc.stroke(
-            Line::new(
-                Point { x, y: 0.0 },
-                Point {
-                    x,
-                    y: canvas_height,
-                },
-            ),
-            &Color::WHITE,
-            1.0,
-        );
-        // lane markings
-        if let Lane::Travel {
-            direction: Some(direction),
-            ..
-        } = lane
-        {
-            let x =
-                (grassy_verge + asphalt_buffer) + (idx as f64 * lane_width) + (0.5 * lane_width);
-            draw_arrow(
-                rc,
-                Point {
-                    x,
-                    y: 0.3 * canvas_height,
-                },
-                *direction,
-            )?;
-            draw_arrow(
-                rc,
-                Point {
-                    x,
-                    y: 0.7 * canvas_height,
-                },
-                *direction,
-            )?;
+
+    let mut left_edge = grassy_verge + asphalt_buffer;
+
+    for lane in &road.lanes {
```

Comment:
Cool, this left-to-right rendering is much closer to what A/B Street and Streetmix and such do! This shows the power of treating markings as a lane, very nice